### DISCUSSION TITLE
Fenced code block not rendering properly

### DISCUSSION THREAD

**Discussion by lnfel** [spatie/laravel-markdown#22] [Q&A] [answered]
I used spatie/laravel-markdown a few months back and got it working perfectly for our company documentation. After sometime I took the project again and to my horror the code blocks are not rendered properly.

I'm using laravel 8 and issue is happening across different laravel-markdown versions,  1.1.1, 1.1.3 and latest 2.0.1
I've read on #1 that this might be an issue of `getenv('PATH')` not returning anything and cannot find `node` binary but `getenv('PATH')` does return path to `node`, my node version is 15.8.0. Markdown config is also set to default.

Expected:
Blocks surrounded with triple backticks should be wrapped around with a `pre` tag
```
something like this
```
Output html is like below:
" something like this "

I even tried to whip up a new laravel project and output is the same without any pre tags
1. `composer require spatie/laravel-markdown`
2. `npm install shiki`
3. in `routes/web.php`
```
Route::get('/', function() {
	return view('test');
});
```
4. in `test.blade.php`
```
<x-markdown>
# Markdown

[Link](#link)

//``` of course without the double slash
code block
not working
//```
</x-markdown>
```
Is there anything that I have missed? I just read the docs again and followed the same steps.

**Comment by lnfel** [comment, marked as answer]
Finally I resorted to using laravel built in markdown:
```php
// in my controller

return view('docs.page', [
    'markdown' => Str::markdown(file_get_contents($docFile))
]);
```

```blade
// in my view layout
// the prose class here from tailwind typography helps in styling the markdown
// specially the fenced code blocks

<main class="px-8 xl:px-16 py-8 prose">
    @yield('content')
</main>
```

```blade
// in docs.page view

@extends('layouts.doc')

@section('pageTitle', 'Developer Documentation')

@section('content')
  {!! $markdown !!}
@endsection
```

With this setup we already got the markdown with correct tags. But we are still missing some Highlighting. 

```html
// going back to view layout

<script src="https://unpkg.com/shiki@0.9.14/dist/index.unpkg.iife.js"></script>
<script>
    (function() {
        // use shikijs highlighter here
        var codeBlock = document.querySelectorAll('pre')
        codeBlock.forEach(function(el) {
            shiki.getHighlighter({
                theme: 'material-palenight',
            }).then(highlighter => {
                var code = highlighter.codeToHtml(el.innerText, 'bat');

                // using my custom helper function to insert newly created dom node from string
                insertAfter(el, htmlToElement(code));
                // remove the previous code block
                el.parentNode.removeChild(el);
            });
        })
    })();
</script>
```

Note: shiki is really slow if using the cdn because it loads a bunch of grammar files but this is the minimal setup to get the code block highlighted

**Reply by makowskid** [reply]
Doesn't work for me either :-/
